src/lib/transaction_logic/zkapp_command_logic.ml

```diff
@@ -1603,7 +1603,11 @@ module Make (Inputs : Inputs_intf) = struct
       let delta_settled =
         Amount.Signed.equal local_state.excess Amount.(Signed.of_unsigned zero)
       in
-      Bool.((not is_last_account_update) ||| delta_settled)
+      let is_empty_account_update =
+        Bool.(is_start' &&& is_last_account_update)
+      in
+      Bool.(
+        is_empty_account_update ||| not is_last_account_update ||| delta_settled)
```

Comment:
I think it is sufficient to check if it is `is_start`, right? So the condition here would be `is_start ||| not is_last_account_update ||| delta_settled`- 
i) ignore local excess if it `is_start` because it will be promoted to global excess and then set to zero later in the code
ii) ignore everything but last account update since the excess wouldn't have been settled
iii) Excess should be settled after the last account update has been applied.

(please add this comment to the code if it looks correct to you)